### DISCUSSION TITLE
DataGrid with duplicate items

### DISCUSSION THREAD

**Discussion by rgomezia** [Megabit/Blazorise#5599] [General]
Hi people , i try to  work with duplicate items but the grid automatically update the similar items when i edit someone or when add a new item i have a method this give LineId automatically , but the same problem when add other similar items all remain with the lineid of the last one

a have this in code lineId Method and Grind Config.

`
ordenMantetimiento.VID_MTOMMACollection = new List<VIDMTOMMACollection>();
for (int i = 0; i < currentList.Count; i++)
{
    currentList[i].LineId = i + 1;
}
ordenMantetimiento.VID_MTOMMACollection = currentList;`

`
     <DataGrid @ref="dataGridMateriales"
     TItem="VIDMTOMMACollection"
     Data="@ordenMantetimiento.VID_MTOMMACollection.Where(x=> x.U_ItemName != string.Empty)"
     Sortable="true"
     Filterable="false"
     ShowPager="false"
     Responsive="true"
     Editable
     EditMode="DataGridEditMode.Inline"
     RowSelectable=@((x)=> x.Item.U_Cerrada != "Y")
     Narrow="true"
     VirtualizeOptions="@(new() { DataGridHeight = "350px"})"
     Virtualize>
`

Plz if someone know how to resolve , that help me a lot. TY

**Comment by David-Moreira** [comment]
What-s in the currentList? Are you sure all the items are different object references? 
We'll need a full-code reproducible, I can't see anything else wrong with just that bit of code.

**Comment by rgomezia** [comment]
I was able to solve it, I really don't know why but when passing the context as a parameter to my method and then adding it to my list it didn't take it, I had to manually create the obj and map it like this.

            int nextLineId = 1;
            foreach (var item in ordenMantetimiento.VID_MTOMMACollection)
            {
                item.LineId = nextLineId++;
            }

            var material = new VIDMTOMMACollection
                {
                    DocEntry = ordenMantetimiento.DocEntry,
                    VisOrder = 1,
                    Object = "VID_MTOM",
                    U_WhsCode = tallerSeleted.U_WhsCode,
                    U_ItemCode = materialSelected.U_ItemCode,
                    U_ItemName = materialSelected.U_ItemName,
                    LineId = nextLineId++
                };

            ordenMantetimiento.VID_MTOMMACollection.Add(material);`

**Comment by stsrki** [comment]
Is `VIDMTOMMACollection `a record or struct type?
